### ISSUE TITLE
ordinal 345 error first time launching application

### ISSUE THREAD

**Issue by sammilucia** [snowie2000/mactype#716]
@snowie2000 Thanks for your work very much.
I recieve the error msg of "oridinal 345" every time when electron apps like terminus and typora launch for the first time and suspect that slack's random crash is caused by Mactype.
For me, this error is definitely reproduceable as long as the system is rebooted. However, VScode never crashes.

After using your latest build [here](https://github.com/snowie2000/mactype/files/5427074/MacType64.zip), the same error msg still show up.
The error msg goes away if Mactype is not loaded automatically and show up once Mactype is loaded at least in Service Mode.

The OS is Windows 10 20H2 with the latest patch. Feel free to let me know if you need additional information.

_Originally posted by @Mint-M in https://github.com/snowie2000/mactype/issues/488#issuecomment-756121873_

**Comment by sammilucia** [comment]
hi @Mint-M, thanks for reporting.

this is due to some other process locking a DLL at the same time an application is trying to use it. usually this is a problem with antivirus, please try disabling your antivirus temporarily and see if the problem still occurs and report back.

thank you 😊

**Comment by MintMana** [comment]
Appreciate this information very much. @sammilucia 

Symantec Endpoint Protection is installed as you suspect and is very possibly to cause this issue. With Eset installed in another PC, the same problem has never occurred.
Unfortunately, it doesn't depend on me whether Symantec can be disabled. I have tried added Mactype, comctl32.dll and these electron apps themselves to the exception list but it doesn't help.

Since this issue only happens on electron apps, do you have any idea about the root cause beyond the antivirus software?

**Comment by sammilucia** [comment]
You could ask your IT people to disable Symantec as a test...

What you can also possibly do is copy the DLLs you're getting errors about to another location in the path. Windows will usually search the program install directory, as well as C:\Windows\System32 and a few other places I can't remember. I think if you use a tool called FileMon it will show you every path that's checked.

You can copy the DLL to another path that's being checked. The idea is, if it can't load the DLL from one location because it's locked, it may be able to load it from another. Doing this is not good practice though, because you'll end up with different versions of the DLL in different places... It's can be useful for testing though.

Probably a better option is just to uninstall MacType and see if you still have the same problem. If you do, it's highly likely it's Symantec, and you can lodge a problem ticket with your IT people. If you DON'T have the problem, then we can help you exclude some apps from MacType.

Really though, MacType loads into DLLs but it doesn't ever lock them.

Let us know how you go? 😊

**Comment by MintMana** [comment]
Yeah, I'm actually asking the IT team about it, however, the configuration of Symantec is applied from a central server just like how GPOs are applied via Windows AD. That is to say, the IT team is not supposed to disable Symantec manually on my PC and the ticket created will take a long time to resolve because even the security department will participate in it.

Judging from the err msg, I have no idea what dlls to copy as you suggested and the msg is similar to the one in https://github.com/snowie2000/mactype/issues/702 which contains no specific dll name. I tried to copy comctl32.dll which @snowie2000 mentioned but it still didn't work.
![Snipaste_2021-01-09_13-33-16](https://user-images.githubusercontent.com/47080894/104086692-f12fa100-5294-11eb-8b45-30354e6343cc.png)

I'm pretty sure that this issue is related to Mactype because if Mactype is loaded manually, the same problem never occurs. In contrast, the err msg shows up for every fresh system boot when Mactype is loaded as Service Mode. Mactype probably doesn't trigger this issue as you said, but Symantec, Mactype and electron apps don't seem work well together.

For disabling Symantec, I'm sorry that it can't be disabled right now but will update for any information once I'm able to run Mactype without it.

Many thanks for following up.

**Comment by sammilucia** [comment]
Hmm, yes, I'm thinking maybe it's the combination of all 3, too. Maybe MacType loading into the DLL is triggering the AV to scan. Are you able to exclude the whole MacType folder and Typora folder in Symantec? If both are excluded there should be no reason to lock the DLL.

Another problem is that AV software doesn't always honor exclusions. But hopefully it works.

**Comment by makazeu** [comment]
This issue is occurring to me as well.
